specification/security/resource-manager/Microsoft.Security/preview/2022-12-01-preview/defenderForStorageSettings.json

```diff
@@ -0,0 +1,257 @@
+          },
+          {
+            "$ref": "#/parameters/DefenderForStorageSetting"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create or update Defender for Storage settings.",
+            "schema": {
+              "$ref": "#/definitions/DefenderForStorageSetting"
+            }
+          },
+          "201": {
+            "description": "Request to  create or update Defender for Storage settings partially succeeded.",
+            "schema": {
+              "$ref": "#/definitions/DefenderForStorageSetting"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DefenderForStorageSetting": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DefenderForStorageSettingProperties"
+        }
+      },
+      "description": "The Defender for Storage resource."
+    },
+    "DefenderForStorageSettingProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
+          "description": "Indicates whether Defender for Storage is enabled on this storage account."
+        },
+        "malwareScanning": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/MalwareScanningProperties"
+        },
+        "sensitiveDataDiscovery": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/SensitiveDataDiscoveryProperties"
+        },
+        "overrideSubscriptionLevelSettings": {
+          "type": "boolean",
+          "description": "Indicates whether the settings defined for this storage account should override the settings defined for the subscription."
+        }
+      },
+      "description": "Defender for Storage resource properties."
+    },
+    "MalwareScanningProperties": {
+      "type": "object",
+      "properties": {
+        "onUpload": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/OnUploadProperties"
+        },
+        "operationStatus": {
+          "type": "object",
+          "$ref": "#/definitions/OperationStatus",
+          "readOnly": true,
+          "description": "Upon failure or partial success. Additional data describing Malware Scanning enable/disable operation."
+        }
+      },
+      "description": "Properties of Malware Scanning."
+    },
+    "OnUploadProperties": {
+      "type": "object",
+      "properties": {
+        "isEnabled": {
+          "type": "boolean",
```

Comment:
Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important